### PR TITLE
feat(client): Add methods for working with the "devices" API.

### LINKED ISSUE #176: Add "devices" API methods
The endpoints are listed in https://github.com/mozilla/fxa/blob/phil/fxa-45-requirements/features/FxA-45-device-registration-api/README.md

Part of https://github.com/mozilla/fxa-content-server/issues/3250

### PR DESCRIPTION
**Pull request by shane-tomlinson**
fixes #176

Early stage WIP, nowhere even close to being ready for feedback.

### CODE REVIEW

**Comment by shane-tomlinson** [comment]
Pulling the WIP off of this, @philbooth, can you check to see this matches your expectations of how the API should be used, @vladikoff - can you review to see if this matches fxa-js-client conventions and everything else?

**Comment by philbooth** [comment]
From the API perspective, this looks perfect.

File: client/FxAccountClient.js [hunk 1]
```diff
@@ -94,6 +95,30 @@ define([
           var requestOpts = {};
 
           if (options) {
+            if (options.device) {
+              requestOpts.device = {};
```
Comment (vladikoff) [inline client/FxAccountClient.js]: Looks like options.device is not documented in the doc string.
It is not part of the 
```
    *   @param {String} [options.resume]
    *   Opaque url-encoded string that will be included in the verification link
    *   as a querystring parameter, useful for continuing an OAuth flow for
    *   example.
    *   @param {String} [options.lang]
``` 

doc blob

File: client/FxAccountClient.js [hunk 2]
```diff
@@ -851,6 +900,123 @@ define([
...
+        return request.send('/account/device', 'POST', creds, data);
+      });
+  };
+
+  /**
+   * Update the name of an existing device
+   *
+   * @method deviceRegister
+   * @param {String} sessionToken User session token
+   * @param {String} id of device
```
Comment (vladikoff) [inline client/FxAccountClient.js]: `id of device` => `deviceId id of device`

File: client/FxAccountClient.js [hunk 3]
```diff
@@ -851,6 +900,123 @@ define([
...
+        return request.send('/account/device', 'POST', creds, data);
+      });
+  };
+
+  /**
+   * Unregister an existing device
+   *
+   * @method deviceDestroy
+   * @param {String} sessionToken sessionToken obtained from signIn
+   * @param {String} id of device
```
Comment (vladikoff) [inline client/FxAccountClient.js]: `id of device` => `deviceId id of device`

**Comment by vladikoff** [comment]
some nits, looks good otherwise!

**Comment by philbooth** [comment]
@vladikoff, I took the liberty of pushing some fixes to your comments on this one. Take another look then merge if you're happy!

**Pull request merged**